cmd/nodepool/core/create.go

```diff
@@ -0,0 +1,113 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/util"
+	hyperapi "github.com/openshift/hypershift/support/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CreateNodePoolOptions struct {
+	Name         string
+	Namespace    string
+	ClusterName  string
+	NodeCount    int32
+	ReleaseImage string
+	Render       bool
+}
+
+type PlatformOptions interface {
+	// UpdateNodePool is used to update the platform specific values in the NodePool
+	UpdateNodePool(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster, client crclient.Client) error
+	// Type returns the platform type
+	Type() hyperv1.PlatformType
+}
+
+func (o *CreateNodePoolOptions) CreateNodePool(ctx context.Context, platformOpts PlatformOptions) error {
+	client := util.GetClientOrDie()
+
+	hcluster := &hyperv1.HostedCluster{}
+	err := client.Get(ctx, types.NamespacedName{Namespace: o.Namespace, Name: o.ClusterName}, hcluster)
+	if err != nil {
+		return fmt.Errorf("failed to get HostedCluster %s/%s: %w", o.Namespace, o.Name, err)
+	}
+
+	if platformOpts.Type() != hcluster.Spec.Platform.Type {
+		return fmt.Errorf("NodePool platform type %s must be HostedCluster type %s", platformOpts.Type(), hcluster.Spec.Platform.Type)
+	}
+
+	nodePool := &hyperv1.NodePool{}
+	err = client.Get(ctx, types.NamespacedName{Namespace: o.Namespace, Name: o.Name}, nodePool)
+	if err == nil && !o.Render {
+		return fmt.Errorf("NodePool %s/%s already exists", o.Namespace, o.Name)
+	}
+
+	var releaseImage string
+	if len(o.ReleaseImage) > 0 {
+		releaseImage = o.ReleaseImage
+	} else {
+		releaseImage = hcluster.Spec.Release.Image
+	}
+
+	nodePool = &hyperv1.NodePool{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "NodePool",
+			APIVersion: hyperv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: o.Namespace,
+			Name:      o.Name,
+		},
+		Spec: hyperv1.NodePoolSpec{
+			Management: hyperv1.NodePoolManagement{
+				UpgradeType: hyperv1.UpgradeTypeReplace,
+			},
+			ClusterName: o.ClusterName,
+			NodeCount:   &o.NodeCount,
+			Release: hyperv1.Release{
+				Image: releaseImage,
+			},
+			Platform: hyperv1.NodePoolPlatform{
+				Type: hcluster.Spec.Platform.Type,
+			},
+		},
+	}
+
+	if err := platformOpts.UpdateNodePool(ctx, nodePool, hcluster, client); err != nil {
+		return err
+	}
+
+	if o.Render {
+		err := hyperapi.YamlSerializer.Encode(nodePool, os.Stdout)
+		if err != nil {
+			panic(err)
+		}
+		return nil
+	}
+
+	var nodePoolBytes bytes.Buffer
+	err = hyperapi.YamlSerializer.Encode(nodePool, &nodePoolBytes)
+	if err != nil {
+		return err
+	}
+
+	if o.Render {
```

Comment:
how is this code run? `if o.Render {` wouldn't this func return in l92?

Reply:
You right, this is never called
This code was already exists before my change
It seems that the "Render" code (writing bytes to `os.Stdout`) was written twice and the second call is unnecessary and can be removed
Am I right?